builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ImportImageConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	imgregv1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ImportTargetKind       = "ContentLibrary"
+	ImportTargetAPIVersion = "imageregistry.vmware.com/v1alpha1"
+
+	DefaultWatchImageImportTimeoutSec = 600
+
+	ImportRequestDefaultNamePrefix    = "packer-vsphere-supervisor-import-req-"
+	StateKeyImageImportRequestCreated = "item_import_req_created"
+)
+
+var IsWatchingImageImport bool
+
+type ImportImageConfig struct {
```

Comment:
It would be nice if we could also provide an option to keep the import source image (true by default). If set to false, it would allow users to clean up the imported image after deploying a VM from it.

Reply:
Hm... it could be useful, I think it doesn't belong there though, it better be in the CreateSource step if to add it. I'll leave it as it is for now, may add the option if I get some cycle to do it.